File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/di/impl/dagger/modules/IssueModule.java
package com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger.modules;

import com.duxtinto.intellij.plugin.github.codereviews.domain.issues.IssuesDomainContract;
import com.duxtinto.intellij.plugin.github.codereviews.ide.acl.services.tasks.IdeaTaskActivator;
import com.duxtinto.intellij.plugin.github.codereviews.net.issues.ApiV3IssueFetcher;
import dagger.Binds;
import dagger.Module;

@Module
public abstract class IssueModule {
    @Binds
    public abstract IssuesDomainContract.Switcher provideIssuesDomainContractSwitcher(IdeaTaskActivator switcher);

    @Binds
    public abstract IssuesDomainContract.Fetcher provideIssuesDomainContractFetcher(ApiV3IssueFetcher fetcher);
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/di/impl/dagger/modules/IssueModule.kt
package com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger.modules

import com.duxtinto.intellij.plugin.github.codereviews.domain.issues.IssuesDomainContract
import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.issues.GetAllClosableByInteractor
import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.issues.ParseIssuesFromStringInteractor
import com.duxtinto.intellij.plugin.github.codereviews.ide.acl.services.tasks.IdeaTaskActivator
import com.duxtinto.intellij.plugin.github.codereviews.net.issues.apiv3.IssueFetcher
import dagger.Binds
import dagger.Module

@Module
abstract class IssueModule {
    @Binds
    abstract fun provideGetAllClosableByInteractor(interactor: GetAllClosableByInteractor): ParseIssuesFromStringInteractor

    @Binds
    abstract fun provideIssuesDomainContractSwitcher(switcher: IdeaTaskActivator): IssuesDomainContract.Switcher

    @Binds
    abstract fun provideIssuesDomainContractFetcher(fetcher: IssueFetcher): IssuesDomainContract.Fetcher
}
